kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/dates.kt

```diff
@@ -42,31 +43,47 @@ fun Arb.Companion.period(maxYear: Int = 10): Arb<Period> = arbitrary(listOf(Peri
 fun Arb.Companion.localDate() = Arb.Companion.localDate(LocalDate.of(1970, 1, 1), LocalDate.of(2030, 12, 31))
 
 /**
- * Arberates a stream of random LocalDates
+ * Generates a stream of random [LocalDate] instances within the specified range.
  *
- * This generator creates randomly generated LocalDates, in the range [[minDate, maxDate]].
+ * This generator creates random [LocalDate] values in the inclusive range from [minDate] to [maxDate].
  *
- * If any of the years in the range contain a leap year, the date [29/02/YEAR] will always be a constant value of this
- * generator.
+ * It includes special edge cases for testing purposes:
+ * - If the range includes any leap years, the date **February 29th** of the first such leap year is included.
+ * - If the range includes any century years (years divisible by 100), the date **January 1st** of the first such century year is included.
+ * - If the range includes any millennium years (years divisible by 1000), the date **January 1st** of the first such millennium year is included.
  *
- * @see [localDateTime]
- * @see [localTime]
+ * These edge cases are added to increase the likelihood of covering date-related boundary conditions in tests.
+ *
+ * @param minDate The minimum (earliest) date to generate (inclusive). Default is January 1st, 1970.
+ * @param maxDate The maximum (latest) date to generate (inclusive). Default is December 31st, 2030.
+ * @return An [Arb]<[LocalDate]> that generates dates within the specified range, including edge cases.
+ *
+ * @see localDateTime
+ * @see localTime
  */
 fun Arb.Companion.localDate(
    minDate: LocalDate = LocalDate.of(1970, 1, 1),
    maxDate: LocalDate = LocalDate.of(2030, 12, 31)
-): Arb<LocalDate> = when {
-   minDate > maxDate -> throw IllegalArgumentException("minDate must be before or equal to maxDate")
-   minDate == maxDate -> Arb.constant(minDate)
-   else -> {
-      val leapYears = (minDate.year..maxDate.year).filter { isLeap(it.toLong()) }
+): Arb<LocalDate> {
+   require(minDate <= maxDate) { "minDate must be before or equal to maxDate" }
+   if (minDate == maxDate) return Arb.constant(minDate)
+
+   val edgeCases = mutableListOf(minDate, maxDate)
+
+   val leapYear = (minDate.year..maxDate.year).firstOrNull { isLeap(it.toLong()) }
+   if (leapYear != null) { edgeCases += LocalDate.of(leapYear, 2, 29) }
+
+   val centuryYear = (minDate.year..maxDate.year).firstOrNull { it % 100 == 0 }
```

Comment:
Do we need unit tests to verify that these edge cases are included:
* centuryYear
* millenniumYear
* leapYear

Reply:
Good idea.

We have one for leap year, but none for century and millennium. I'll include them

Reply:
Done. I've also removed the millennium edge cases altogether.

They don't add much value considering we already created the century edge case which will most of the times overlap with the millennium unless under very specific scenarios the first millenium is different from the first century in the date range.